Add app loader

eatonphil opened the pull request:
<img width="642" alt="screen shot 2017-05-24 at 10 13 47 am" src="https://cloud.githubusercontent.com/assets/3925912/26407657/c1243426-4069-11e7-8d00-be393d30a901.png">

abemassry commented:
I'm a fan of a loading screen in general as opposed to showing a blank screen for anything > 4 sec

mparke commented on scss/components/AppLoader.scss:
```diff
@@ -0,0 +1,97 @@
+// Source: https://github.com/lukehaas/css-loaders
```
can be included via npm?

eatonphil replied:
Yes, but it's a pretty small dependency and I needed to make some changes to it.

eatonphil replied:
More importantly, it's not like we need to keep it up-to-date with upstream security / bug fixes sort of thing.

mparke replied:
so, you forked it?

mparke commented:
We could show this over the UI with an opacity:

![screen shot 2017-05-24 at 2 24 37 pm](https://cloud.githubusercontent.com/assets/709951/26419624/53401720-408e-11e7-8d9c-92a647935650.png)

eatonphil commented:
So long as response times are reasonable (not 10 seconds), I'd rather the app be fully ready before we show anything.

phuegel commented:
The header (minus notification counts) should be static. Why can't they be loaded immediately while you wait for the body content below? Like the following (Blue constant & Red dynamic content waiting for complete responses):

![image](https://cloud.githubusercontent.com/assets/19841047/26423560/1f883ec2-409c-11e7-8086-2d69f6d11ba3.png)

phuegel commented:
BTW - see the following for React loading screen example. 
https://stackoverflow.com/questions/40987309/react-display-loading-screen-while-dom-is-rendering

mparke commented:
I agree w/ @phuegel that we should show the header UI at least while body loads.

eatonphil commented:
The most we can do here is display the header with the Linode logo (and the three main nav items). However, it is not possible to show the rest of the items because (like the body), they are not returned immediately. I'll make a commit to add the header and the Linode logo for now. But I don't think it's appropriate to show the header nav items because those are clearly "logged in" items and it isn't clear you aren't entirely "logged in" until we can display your username at least.

eatonphil commented:
Updated with some nice improvements:

<img width="642" alt="screen shot 2017-05-26 at 1 38 28 pm" src="https://cloud.githubusercontent.com/assets/3925912/26505940/a74f56ec-4218-11e7-9c2b-f791197e9e53.png">

Improvements include:

* Moved all session related logic into src/sessions.js alone
* Make authenticated calls first in src/layouts/Layouts.js so the redirect to login gets triggered sooner
* If a request fails with 401, remove the current token from localstorage so if you happen to not log in immediately and refresh the page, no time is wasted redirecting you to login.

mparke commented:
@eatonphil build failing?
